src/js/structs/Job.js

```diff
@@ -0,0 +1,21 @@
+import Item from './Item';
+
+module.exports = class Job extends Item {
+  constructor() {
+    super(...arguments);
+
+    const id = this.getId();
+    if (id !== '/' && (!id.startsWith('/') || id.endsWith('/'))) {
+      throw new Error(`Id (${id}) must start with a leading slash ("/") and should not end with a slash, except for root id which is only a slash`);
```

Comment:
The backend is using the same [PathId](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/state/PathId.scala) as Marathon and as far as I can tell already strips dangling slashs for us. 
In the very rare case we actually receive malformed data we should handle it in the same way we handle malformed JSON. Let's catch these errors in the Chronos store and report them to the user instead of introducing a ton of complex regex patterns in an attempt to fix a broken backend response. 

/cc @gkleiman